server/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -392,14 +405,43 @@ public static Processor readProcessor(Map<String, Processor.Factory> processorFa
                         type, Arrays.toString(config.keySet().toArray()));
                 }
                 if (onFailureProcessors.size() > 0 || ignoreFailure) {
-                    return new CompoundProcessor(ignoreFailure, Collections.singletonList(processor), onFailureProcessors);
-                } else {
-                    return processor;
+                    processor = new CompoundProcessor(ignoreFailure, Collections.singletonList(processor), onFailureProcessors);
                 }
+                if (conditionalScript != null) {
+                    processor = new ConditionalProcessor(tag, conditionalScript, scriptService, processor);
+                }
+                return processor;
             } catch (Exception e) {
                 throw newConfigurationException(type, tag, null, e);
             }
         }
         throw newConfigurationException(type, tag, null, "No processor type exists with name [" + type + "]");
     }
+
+    private static Script maybeExtractConditional(Map<String, Object> config) throws IOException {
```

Comment:
since you are returning the script, I think this can just be called `extractConditional`